hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/execution/bulkinsert/RDDSpatialCurveOptimizationSortPartitioner.java

```diff
@@ -62,16 +61,12 @@ public RDDSpatialCurveOptimizationSortPartitioner(HoodieSparkEngineContext spark
 
   @Override
   public JavaRDD<HoodieRecord<T>> repartitionRecords(JavaRDD<HoodieRecord<T>> records, int outputSparkPartitions) {
-    String payloadClass = config.getPayloadClass();
-    // do sort
     JavaRDD<GenericRecord> preparedRecord = prepareGenericRecord(records, outputSparkPartitions, serializableSchema.get());
     return preparedRecord.map(record -> {
       String key = record.get(HoodieRecord.RECORD_KEY_METADATA_FIELD).toString();
       String partition = record.get(HoodieRecord.PARTITION_PATH_METADATA_FIELD).toString();
       HoodieKey hoodieKey = new HoodieKey(key, partition);
-      HoodieRecordPayload avroPayload = ReflectionUtils.loadPayload(payloadClass,
-          new Object[] {Option.of(record)}, Option.class);
-      HoodieRecord hoodieRecord = new HoodieRecord(hoodieKey, avroPayload);
+      HoodieRecord hoodieRecord = new HoodieRecord(hoodieKey, new RewriteAvroPayload(record));
```

Comment:
this is okay, as long as the new base and log files have been merged already in `records`. I think this is the case. Could you confirm again? the case for using the user-defined payload class for clustering, is when the clustering also implicitly compacts the base and log files.

Reply:
Correct. This is done by `MultipleSparkJobExecutionStrategy.readRecordsForGroup` method prior to calling into Partitioner